instrumentation/kafka-clients/kafka-clients-0.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kafkaclients/TracingIterator.java

```diff
@@ -42,6 +43,14 @@ public TracingIterator(
     this.parentContext = parentContext;
   }
 
+  public static <K, V> Iterator<ConsumerRecord<K, V>> wrap(
+      Iterator<ConsumerRecord<K, V>> delegateIterator, @Nullable SpanContext receiveSpanContext) {
+    if (KafkaTracingWrapperUtil.wrappingEnabled()) {
+      return new TracingIterator<>(delegateIterator, receiveSpanContext);
+    }
+    return delegateIterator;
+  }
+
```

Comment:
I think there's a similar use case that @mateuszrzeszutek recently ran into recently:

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/be090e431a39f54c8d909ac74838ed0a19818322/instrumentation/kafka-streams-0.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kafkastreams/StreamTaskInstrumentation.java#L74-L78

would it work to use the same "workaround" for both? (either thread local suppression, or force unwrapping for both)

Reply:
I think I can use thread local for both, I'll just have to move the thread local to a different method that exists in all kafka-streams versions